geo/src/algorithm/geodesic_area.rs

```diff
@@ -0,0 +1,504 @@
+    /// For a polygon this returns the sum of the perimeter of the exterior ring and interior rings.
+    /// To get the perimeter of just the exterior ring of a polygon, do `polygon.exterior().geodesic_length()`.
+    ///
+    /// # Units
+    ///
+    /// - return value: (meter, meter²)
+    /// 
+    /// # Interpreting negative area values
+    /// 
+    /// A negative area value can mean one of two things:
+    /// 1. The winding of the polygon is in the clockwise direction (reverse winding). If this is the case, and you know the polygon is smaller than half the area of earth, you can take the absolute value of the reported area to get the correct area.
+    /// 2. The polygon is larger than half the planet. In this case, the returned area of the polygon is not correct. If you expect to be dealing with very large polygons, please use the 'unsigned' methods.
+    /// 
+    /// [Karney (2013)]:  https://arxiv.org/pdf/1109.4448.pdf
+    fn geodesic_perimeter_area_signed(&self) -> (T, T);
+
+    /// Determine the perimeter and area of a geometry on an ellipsoidal model of the earth, all in one operation. Supports very large geometries that cover a significant portion of the earth.
+    /// 
+    /// This returns the perimeter and area in a `(perimeter, area)` tuple and uses the geodesic measurement methods given by [Karney (2013)].
+    /// 
+    /// # Area Assumptions
+    ///  - Polygons are assumed to be wound in a counter-clockwise direction 
+    ///    for the exterior ring and a clockwise direction for interior rings. 
+    ///    This is the standard winding for Geometries that follow the Simple Features standard. 
+    ///    Using alternative windings will result in incorrect results.
+    /// 
+    /// # Perimeter
+    /// For a polygon this returns the perimeter of the exterior ring and interior rings.
+    /// To get the perimeter of just the exterior ring of a polygon, do `polygon.exterior().geodesic_length()`.
+    ///
+    /// # Units
+    ///
+    /// - return value: (meter, meter²)
+    /// 
+    /// [Karney (2013)]:  https://arxiv.org/pdf/1109.4448.pdf
+    fn geodesic_perimeter_area_unsigned(&self) -> (T, T);
+}
+
+impl GeodesicArea<f64> for Polygon {
+    fn geodesic_perimeter(&self) -> f64 {
+        let (perimeter, _area) = geodesic_area(self, true, false, false);
+        perimeter
+    }
+
+    fn geodesic_area_signed(&self) -> f64 {
+        let (_perimeter, area) = geodesic_area(self, true, false, false);
+        area
+    }
+
+    fn geodesic_area_unsigned(&self) -> f64 {
+        let (_perimeter, area) = geodesic_area(self, false, false, false);
+        area
+    }
+
+    fn geodesic_perimeter_area_signed(&self) -> (f64, f64) {
+        geodesic_area(self, true, false, false)
+    }
+
+    fn geodesic_perimeter_area_unsigned(&self) -> (f64, f64) {
+        geodesic_area(self, false, false, false)
+    }
+}
+
+fn geodesic_area(poly: &Polygon, sign: bool, reverse: bool, exterior_only: bool) -> (f64, f64) {
```

Comment:
`reverse` and `exterior_only` seems to always be false. Do you want to get rid of them? Or if you want to keep them, can you comment as to their future purpose?

Reply:
Originally I was planning on more features, but decided to reign it in a bit. So I guess some open questions:

1. Do we want to *fully* support clockwise-wounded geometries?  The Documentation (https://docs.rs/geo/latest/geo/#semantics) states "The geospatial types provided here aim to adhere to the [OpenGIS Simple feature access](https://www.ogc.org/standards/sfa) standards."  This suggests to me that we don't really need to *actually* support reverse-wound geometries other than on a minimal "best effort" basis.  If this is true, I'll remove the "reverse" option here to only fully support counterclockwise-wound geometries as per the Simple Feature spec. 

2. Is it useful to have an option to only get the perimeter of the exterior?  Originally I thought this might be useful, but in retrospect just calling `geodesic_length()` on the exterior ring does the same thing.   The only trickly thing here is that sometimes I've seen polygons constructed where the first point of the exterior ring and the last point of the exterior ring are not the same (there's an *implicit* closure of the ring, instead of an explicit closure by duplicating the origin point).   If this case calling `geodesic_length()` on the exterior-ring would result in an incorrect result, but we could correct for that here knowing that the exterior-rings and interior-rings should always close, even if only implicitly.